pkg/distributor/distributor.go

```diff
@@ -281,6 +282,11 @@ func newPushMetrics(reg prometheus.Registerer) *PushMetrics {
 			NativeHistogramMinResetDuration: 1 * time.Hour,
 			NativeHistogramMaxBucketNumber:  100,
 		}, []string{"user"}),
+		otlpIncomingSamplesPerBatch: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
```

Comment:
Unfortunately this will create a lot of series (number of tenants using OTLP * number of distributors * (number of buckets + 2)). It's around 700k series altogether in our production environments. If we used native histograms only, it would be 1/10th of that.

If we go with native histograms, I'd recommend using factor=2, 4 or 16 (buckets would be have boundaries that are exponents of the factor).

Reply:
after discussed offline with Peter, I am going to change factor to 2 and leave the default max bucket which is 160.